exchanges/stream/websocket.go

```diff
@@ -1009,3 +996,46 @@ func (w *Websocket) checkSubscriptions(subs []subscription.Subscription) error {
 
 	return nil
 }
+
+// listen listens to the websocket connection and handles incoming data
+func (w *Websocket) listen(conn Connection, handler func(incoming []byte) error) {
+	defer w.Wg.Done()
+	for {
+		resp := conn.ReadMessage()
+		if resp.Raw == nil {
+			return
+		}
+		if err := handler(resp.Raw); err != nil {
+			w.DataHandler <- fmt.Errorf("websocket streaming connection %s %s: %w", w.exchangeName, conn.GetURL(), err)
```

Comment:
Suggest that along with adding conn Type flag, that's what we'd use here, maybe stringified, but maybe as a structured error, too.
I think the first part of this isn't clear.
```
errWsHandler = errors.New("error in WS data handler")
```
```
"%s: %w %s: %w", eName, errWsHandler, conn.Type, err
```
`bybit: error in WS data handler public: semi-pornographic ascii art received instead of order confirmation`
or maybe as far as
```
package websocket
type HandlerError struct {
 Exchange exchange.IBotExchange (pointer obvs)
 Connection 
 Message ( the thing that caused this )
 Error
}